collagraph/component.py

```diff
@@ -81,3 +82,38 @@ def add_event_handler(self, event, handler):
     def remove_event_handler(self, event, handler):
         """Removes an event handler for the given event."""
         self._event_handlers[event].remove(handler)
+
+    def _lookup(self, name):
+        """
+        Helper method that is used in the template.
+        The method looks up variables that are mentioned in the template.
+        This provides some syntactic sugar so that users can leave out `self`,
+        `self.state` and `self.props`.
+        """
+        cache = Component.__lookup_cache__
```

Comment:
are you sure about putting this on the class? seems like it should be on self...? could be different for every instance right?

Reply:
Yes, I'm sure. This was actually what was happening before as well, but I guess I've removed the comment that accompanied the behaviour...

The idea is that once you've looked up a name, then the lookup is cached. This should be the same for every instance.
This is all under the assumption that there is no overlap of keys in `self.props`, `self.state` and attributes on self.

I guess that we could do with some more runtime warnings on overlapping keys/attributes...

Reply:
OK!